### PR TITLE
[LaTeX] Math scopes

### PR DESCRIPTION
**Pull request by ngc92**
This PR adapts the scoping for maths in TeX/LaTeX

* use same scopes as in markdown
* `=` sign in maths is rarely assingment
* `indice` is not an English word. Changed to `subscript` Q: should `^` be scoped `superscript` for consistency, or do we keep its mathematical meaning of exponentiation?
* scope braces `\{\}` more consistent with other (math-)grouping mechanisms `()` and `[]` 

Q: How should we handle scoping regarding the surrounding construct and the actual contents, i.e.
`$x$`, here only `x` is actually maths. This is currently somewhat inconsistent between the long `begin` and `end` forms, where these commands are not part of the meta environment, but for short forms the begin/end delimiters are taken into account.

### CODE REVIEW

**Comment by ngc92** [comment]
One more question, should we consider math commands that produce such as `\alpha` to be (named) escape sequences and scope them like that, at least with an additional scope?

**Comment by jwortmann** [comment]
I guess removing the `string.other.math` scope for the math delimiters would "break" popular LaTeX plugins and probably a lot of color schemes too.

https://github.com/SublimeText/LaTeXTools/search?q=%22string.other.math%22
https://github.com/LaTeXing/LaTeXing/search?q=%22string.other.math%22

The string scope doesn't make sense semantically, but it might be something that is difficult / infeasable to change now, due to historic reasons and backwards compatibility.

> Q: should `^` be scoped `superscript` for consistency, or do we keep its mathematical meaning of exponentiation?

Yes, imo it should use a similar scope like `_`. It is `keyword.operator.math` on the master branch, and seems to have been adjusted to `keyword.operator.arithmetic` in one of the recent PRs, but this is wrong because it doesn't necessarily mean exponentation, for example consider $A^T$. Okay, "operator" would still fit in this case, but there are a lot of other notations in math or physics where an index is used in form of a superscript.

> * scope braces `\{\}` more consistent with other (math-)grouping mechanisms `()` and `[]`

`\{` is escaped and I think therefore it should be highlighted with the `constant.character.escape` color by default, while `(` or `[` are not escaped. I'm also not sure whether the `punctuation.section` scopes which can be found in this diff (not on master branch) can really be applied to the parentheses or braces like they are in other programming languages, because in LaTeX they are just printed characters and opening & closing parentheses don't necessarily need to match. But the `constant.character.{parenthesis|bracket}` on master is also a bit strange from the highlighting perspective (not sure if this was already discussed in one of the recent PRs).

> One more question, should we consider math commands that produce such as `\alpha` to be (named) escape sequences and scope them like that, at least with an additional scope?

I would say no, assuming you refer to the `constant.character.escape` scope. Otherwise, following that logic, you would need to do the same for all other macros which print as a single character. For example `\partial`, `\nabla`, `\infty` etc.

**Comment by ngc92** [comment]
@jwortmann 
Thanks for the detailed comments.

I think you're right that removing `string.other.math` is too optimistic. It would be nice to have a mechanism so that these could be fixed in the syntax file, i.e. have an additional `scope-remapping` that would add the backward-compatible but semantically questionable scopes based on the actual, sensible scopes. That would also be very helpful in fixing the inconsistencies in noted in #3607.

I think highlighting parenthesis, and especially matching parenthesis, is a valuable help when typing math formulas. There are certain cases where things can go wrong, e.g. if you denote open intervals by `]a, b[` the highlighter will be very confused. Also, `\{\}` currently receives two scopes, one for the escape and one for the grouping mechanism. I think it makes sense to switch the order of the two scopes, though.

**Comment by ngc92** [comment]
I rebased this on the other recent changes, #3787, and also reverted the change to the string scope name

File: LaTeX/TeX.sublime-syntax [hunk 1]
```diff
@@ -677,10 +681,14 @@ contexts:
       scope: punctuation.section.brackets.begin.tex
     - match: \]
       scope: punctuation.section.brackets.end.tex
-    - match: (\\)[{}]
-      scope: constant.character.escape.tex
+    - match: (\\){
```
[the remark this answers is not in the document]
Comment (ngc92) [reply LaTeX/TeX.sublime-syntax:684]: why would I need to escape { here?

[the remark this answers is not in the document]
Comment (FichteFoll) [reply LaTeX/TeX.sublime-syntax:684]: Resolving these as there're very optional and I don't want to block PRs on trivialities like this.

File: LaTeX/TeX.sublime-syntax [hunk 2]
```diff
@@ -677,10 +681,14 @@ contexts:
...
+      scope: punctuation.section.braces.begin.tex constant.character.escape.tex
       captures:
-        1: punctuation.definition.escape.tex
+        1: punctuation.definition.backslash.tex
+    - match: (\\)}
```
[the remark this answers is not in the document]
Comment (FichteFoll) [reply LaTeX/TeX.sublime-syntax:688]: Resolving these as there're very optional and I don't want to block PRs on trivialities like this.

**Review by FichteFoll — APPROVED** [review]

**Pull request merged**